rankers/BiMatchRanker/__init__.py

```diff
@@ -11,12 +11,13 @@ class BiMatchRanker(Chunk2DocRanker):
 
     .. warning:: Here we suppose that the smaller chunk score means the more similar.
     """
-    required_keys = {'length'}
+    query_required_keys = {'length'}
```

Comment:
For BiMatchRanker let it be handled in #4237

Reply:
So I will remove all changes on bimatchranker and only keep the simpleappregrate one?